File: scripts/compress-images.py
"""Compress OTI images using Pillow (resize + quality)."""
import sys, pathlib

try:
    from PIL import Image
except ImportError:
    sys.exit(1)

BASE = pathlib.Path(__file__).parent.parent / "public/assets/img"
FILES = ["bg-municipal.jpg", "OTI.jpeg"]
MAX_W  = 1920
QUALITY = 70

for name in FILES:
    src = BASE / name
    if not src.exists():
        print(f"MISSING {name}")
        continue

    old = src.stat().st_size
    img = Image.open(src)
    w, h = img.size

    try:
        exif = img._getexif() or {}
        orient = exif.get(0x11287, 1)
        if orient == 3: img = img.rotate(180, expand=True)
        elif orient == 6: img = img.rotate(-90, expand=True)
        elif orient == 8: img = img.rotate(90, expand=True)
    except Exception:
        pass

    if w > MAX_W:
        ratio = MAX_W / w
        img = img.resize((MAX_W, int(h * ratio)), Image.LANCZOS)

    img.save(src, "JPEG", quality=QUALITY, optimize=True, progressive=True)
    new = src.stat().st_size
    saved = old - new
    print(f"{name:25} {old//1024:>7} KB -> {new//1024:>7} KB  (saved {saved//1024} KB)")

File: scripts/compress-images.php
<?php
/**
 * Image compression helper — GD bundled in XAMPP
 */
error_reporting(0);

function compressImage(string $src, string $dst, int $quality = 75): void
{
    $info = getimagesize($src);
    if (!$info) { fwrite(STDERR, "Cannot read: $src\n"); return; }

    [$w, $h, $type] = $info;
    $srcImg = match ($type) {
        IMAGETYPE_JPEG => imagecreatefromjpeg($src),
        IMAGETYPE_PNG  => imagecreatefrompng($src),
        default        => null,
    };
    if (!$srcImg) { fwrite(STDERR, "Unsupported format: $src\n"); return; }

    // Resize: max 1920px wide (keeps quality, cuts size 50-70%)
    $maxW = 1920;
    if ($w > $maxW) {
        $ratio = $maxW / $w;
        $newW  = $maxW;
        $newH  = (int)($h * $ratio);
        $tmp   = imagecreatetruecolor($newW, $newH);
        imagecopyresampled($tmp, $srcImg, 0, 0, 0, 0, $newW, $newH, $w, $h);
        imagedestroy($srcImg);
        $srcImg = $tmp;
        [$w, $h] = [$newW, $newH];
    }

    imagejpeg($srcImg, $dst, $quality);
    imagedestroy($srcImg);
}

$base = __DIR__ . '/public/assets/img/';
$files = ['bg-municipal.jpg', 'OTI.jpeg'];

foreach ($files as $f) {
    $src = $base . $f;
    if (!file_exists($src)) { echo "MISSING $f\n"; continue; }
    $oldSize = filesize($src);
    compressImage($src, $src, 70);
    $newSize = filesize($src);
    $saved   = $oldSize - $newSize;
    $pct     = $oldSize > 0 ? round(($saved / $oldSize) * 100, 1) : 0;
    printf("%-25s %8s → %8s  (%-5s KB / %4.1f%%)\n",
        $f,
        number_format($oldSize / 1024, 1) . ' KB',
        number_format($newSize / 1024, 1) . ' KB',
        number_format($saved / 1024, 1) . ' KB',
        $pct
    );
}
